util/stmtsummary/statement_summary.go

```diff
@@ -388,20 +410,63 @@ func (ssMap *stmtSummaryByDigestMap) SetRefreshInterval(value string, inSession
 	}
 }
 
-func (ssMap *stmtSummaryByDigestMap) RefreshInterval() int64 {
+// refreshInterval gets the refresh interval for summaries.
+func (ssMap *stmtSummaryByDigestMap) refreshInterval() int64 {
 	return atomic.LoadInt64(&ssMap.sysVars.refreshInterval)
 }
 
+// SetHistorySize sets the history size for all summaries.
+func (ssMap *stmtSummaryByDigestMap) SetHistorySize(value string, inSession bool) {
+	ssMap.sysVars.Lock()
+	if inSession {
+		ssMap.sysVars.sessionHistorySize = value
+	} else {
+		ssMap.sysVars.globalHistorySize = value
+	}
+	sessionHistorySize := ssMap.sysVars.sessionHistorySize
+	globalHistorySize := ssMap.sysVars.globalHistorySize
+	ssMap.sysVars.Unlock()
+
+	// Calculate the cached `historySize`.
+	var size int
+	var err error
+	if ssMap.isSet(sessionHistorySize) {
+		size, err = strconv.Atoi(sessionHistorySize)
+		if err != nil {
+			size = 0
+		}
+	}
+	if size <= 0 {
+		size, err = strconv.Atoi(globalHistorySize)
+		if err != nil {
+			size = 0
+		}
+	}
+	if size > 0 {
+		atomic.StoreInt32(&ssMap.sysVars.historySize, int32(size))
+	}
+}
+
+// historySize gets the history size for summaries.
+func (ssMap *stmtSummaryByDigestMap) historySize() int {
+	size := int(atomic.LoadInt32(&ssMap.sysVars.historySize))
+	if size <= 0 {
+		size = 1
```

Comment:
Why the default value of `historySize` is 1? Should be 0?

Reply:
or `DefTiDBStmtSummaryHistorySize`?

Reply:
Default value is `DefTiDBStmtSummaryHistorySize`. Just in case that `historySize` is never set.

Reply:
Addressed.